Added Creating Flutter App to Wiki

danielqiuu opened the pull request:
This PR adds a new tech stack, Flutter. It starts off with a brief introduction to Flutter and then walks the reader through the steps to install Flutter onto their machine. After the installation of Flutter, I walk the reader through creating their first Flutter App, which is a first step to cross platform development.

lukecheseldine commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,49 @@
+The following is a summary from the offical [Flutter Docs](https://docs.flutter.dev/get-started/).
+
+1. **Install Flutter**
+   a) [Windows](https://docs.flutter.dev/get-started/install/windows)
```
This displays in-line instead of in a list. To make it more readable, you can use a numbered list, which is natively supported by Markdown. Otherwise, for a lettered list, you can just embed HTML in your file, like this:

```
<ol type="a">
  <li>Coffee</li>
  <li>Tea</li>
  <li>Milk</li>
</ol>
```

Which will compile to this:

<ol type="a">
  <li>Coffee</li>
  <li>Tea</li>
  <li>Milk</li>
</ol>

lukecheseldine commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,49 @@
+
+3. **Create the app**
+   a) Invoke **View > Command Palette**. This can be done with `ctrl shift p` on Windows or `cmd shift p` on Mac.
+   b) Type "flutter", and select the **Flutter: New Project**
```
I think a picture could be very helpful here, especially for people who don't normally use vscode!

lukecheseldine commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,49 @@
+   c) Select **application**
+   d) Create or select the parent directory for the new project folder.
+   e) Enter a project name, such as `my_app`, and press **Enter**.
+   f) Wait for project creation to complete and the `main.dart` file to appear.
```
Maybe you could add an image for what the project structure should look like in the end

lukecheseldine commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,49 @@
+   b) Select a device from the Device Selector area.
+   c) Invoke **Run > Start Debugging** or press `F5`.
+   d) Wait for the app to launch—progress is printed in the Debug Console view.
+   e) After the app build completes, you’ll see the starter app on your device.
```
Attach an image for what it looks like, so the person following the tutorial knows what to expect?

lukecheseldine commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,49 @@
+
+#### [Creating your first flutter app](#creating-your-first-flutter-app)
+
+## What is Flutter?
```
Maybe you could have a section weighing the pros and cons of Flutter compared to other cross-platform frameworks like React Native? Maybe also the drawbacks compared to building native apps with Swift or Kotlin as well!

lukecheseldine reviewed (changes requested):
Great job Daniel! I thought the guide was very easy to follow, and you linked a lot of great tutorials!

Just had a couple of suggestions to improve readability and deepen the discussion points you mention :)

danielqiuu commented:
@lukecheseldine Thank you for the feedback. I agree with adding pictures, especially since VS Code can be hard to navigate. I have addressed your feedback by adding a new section for pros and cons, using the markdown list feature as well as added a few snippets of VS Code that I think might be helpful for the reader. Could you take a look at my PR again when you have the chance?

liu-samuel reviewed (changes requested):
Hi Daniel, I think this is a great guide! When you get the chance, could you take a look at my comments? I just had a couple of suggestions

liu-samuel commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,79 @@
+      <img width="636" alt="VS Code Command Palette" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/65613404/9bdf3ef1-ad9b-4848-b6bb-723f85462c36">
+   2. Type "flutter", and select the **Flutter: New Project**
+      <img width="644" alt="Flutter new project" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/65613404/47cf0a8d-25a5-4076-ad6d-54cc8197384d">
+   3. Select **application**
```
I think this step by step guide is very clear! Luke's suggestion of adding pictures was great

liu-samuel commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,79 @@
+
+## What is Flutter?
+
+Flutter is an open source framework developed and supported by Google. Frontend and full-stack developers use Flutter to build an application’s user interface (UI) for multiple platforms with a single codebase.
```
Could you give some examples of products or companies that use flutter for their development? It would be great to showcase what flutter is capable of

lukecheseldine replied:
+1 on this, great idea!

liu-samuel reviewed (changes requested):
Hi Daniel, I think this is a great guide! When you get the chance, could you take a look at my comments? I just had a couple of suggestions

danielqiuu commented:
@liu-samuel Good catch on the typo! I have addressed your comments in the newest commit so please take a look when you have the chance. I agree that showcasing companies using Flutter is a great way to encourage viewers to consider using it for their own projects.

lukecheseldine commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,82 @@
+
+## Why Flutter? Pros and Cons
+
+####Pros:
```
Put a space after these hashtags!

lukecheseldine commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,82 @@
+3. **Hot Reload Feature**
+   The ability to hot reload is one of the main benefits of using Flutter. This is for effective cross-platform development so it can complement the nature of Flutter. This feature’s function speeds up application development. Hot reloading allows developers to see the changes they make to the code in real time without losing the state of the app. It is useful when editing the UI of an app.
+
+###Cons:
```
Space after the hashtags here too. Also, should it be 3 or 4 hashtags?

lukecheseldine commented on Topics/Tech_Stacks/Flutter.md:
```diff
@@ -0,0 +1,82 @@
+   5. Enter a project name, such as `my_app`, and press **Enter**.
+   6. Wait for project creation to complete and the `main.dart` file to appear.
+      The above commands create a Flutter project directory called my_app that contains a simple demo app that uses [Material Components](https://m3.material.io/components).
+   7. Your file structures should look as follows.
```
The image is being rendered on the same line as the text. Can you add a newline so it looks better?

lukecheseldine reviewed (changes requested):
Great work Daniel! I think the images make a huge difference. Left a couple comments about formatting and markdown mistakes, should be a quick fix :)

liu-samuel reviewed (approved):
This looks good to me! This is an interesting guide and very easy to follow!

lukecheseldine reviewed (approved):
Great job Daniel! LGTM!
